File: communication-microservice/src/main/java/edu/hm/interaction/service/InteractionService.java
package edu.hm.interaction.service;

import edu.hm.interaction.persistence.Interaction;

import java.util.List;
import java.util.Optional;

/**
 * Interface for interaction service.
 *
 * @author Simon Hirner
 */
public interface InteractionService {

    List<Interaction> getAllInteractions();

    Optional<Interaction> getInteraction(String id);

    Interaction saveInteraction(Interaction newInteraction);

    Interaction replaceInteraction(String id, Interaction newInteraction);

    void deleteInteraction(String id);

    void deleteAllInteractions();
}

File: crm-frontend/src/services/InteractionService.js
import axios from "axios";

const BACKEND = process.env.REACT_APP_BACKEND;

const INTERACTION_API = 'http://' + BACKEND + (BACKEND === 'localhost' ? ':8081' : ':30011') + '/interactions';

class InteractionService {

    getAllInteractions() {
        return axios.get(INTERACTION_API);
    }

    saveInteraction(Interaction) {
        return axios.post(INTERACTION_API, Interaction);
    }

    getInteraction(id) {
        return axios.get(INTERACTION_API + "/" + id);
    }

    replaceInteraction(id, Interaction) {
        return axios.put(INTERACTION_API + "/" + id, Interaction);
    }

    deleteInteraction(id) {
        return axios.delete(INTERACTION_API + "/" + id)
    }

}

export default new InteractionService()

